src/figuresClasses.ts

```diff
@@ -2,10 +1,95 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Figure['color'];
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Figure['color'], a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be a positive number');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('This is not a triangle');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Triangle implements Figure {}
+  color: Figure['color'];
 
-export class Circle implements Figure {}
+  r: number;
 
-export class Rectangle implements Figure {}
+  constructor(color: Figure['color'], r: number) {
+    if (r <= 0) {
+      throw new Error('Radius must be a positive number');
+    }
+
+    this.shape = 'circle';
+    this.color = color;
+    this.r = r;
```

Comment:
When you declare a constructor parameter with an access modifier, TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign `color` and `r` to `this`.